File: JobMatchingSystem.API/Models/CVEducation.cs
using JobMatchingSystem.API.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace JobMatchingSystem.API.Models
{
    public class CVEducation
    {
        [Key]
        public int Id { get; set; }
        public int UserId { get; set; }
        public string SchoolName { get; set; }
        public int EducationLevelId { get; set; }
        public string Major { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public string? Description { get; set; }
        // Navigation properties
        [ForeignKey("UserId")]
        public virtual ApplicationUser User { get; set; } = null!;
         [ForeignKey("EducationLevelId")]
        public virtual EducationLevel EducationLevel { get; set; } = null!;
    }
}

File: JobMatchingSystem.FE/src/models/cv-education.ts
export const EducationLevel = {
  1: 'Cao đẳng',
  2: 'Đại học',
  3: 'Kỹ sư',
  4: 'Cử nhân',
  5: 'Thạc sĩ',
  6: 'Tiến sĩ',
} as const;

export type EducationLevel = typeof EducationLevel[keyof typeof EducationLevel];

export interface CVEducation {
  id?: number;
  schoolName: string;
  educationLevelId: number;
  major: string;
  startDate: string;
  endDate: string;
  description: string;
  userId?: number;
  createdAt?: string;
  updatedAt?: string;
}
